js/ui/components/clubhouse.js

```diff
@@ -118,12 +119,30 @@ var ClubhouseNotificationSource = new Lang.Class({
     },
 });
 
+var ClubhouseOpenButton = new Lang.Class({
+    Name: 'ClubhouseOpenButton',
+    Extends: St.Button,
+
+    _init: function(params) {
+        this.parent(params);
+
+        this._icon = new St.Icon({ style_class: 'clubhouse-open-button-icon' });
```

Comment:
You could create the icon before calling `this.parent()` and then assign it to `params.child` to get it set during construction.

Reply:
@cosimoc I tried but it broke my shell with:
```
== Stack trace for context 0x55db51a18650 ==

(gnome-shell:17187): Gjs-CRITICAL **: Attempting to call back into JSAPI during the sweeping phase of GC. This is most likely caused by not destroying a Clutter actor or Gtk+ widget with ::destroy signals connected, but can also be caused by using the destroy(), dispose(), or remove() vfuncs. Because it would crash the application, it has been blocked and the JS callback not invoked.

(gnome-shell:17187): Gjs-CRITICAL **: The offending signal was destroy on StButton 0x55db552f8280.
```

Reply:
@manuq That message does not look like one that would break your shell, and I'm pretty sure it happens regardless of these changes (I see tons of those warnings just running the shell normally)